### DISCUSSION TITLE
Typegen in NX Monorepo Setup

### DISCUSSION THREAD

**Discussion by livingmine** [statelyai/xstate#3476] [Q&A]
Dose anyone know the mechanism behind the creation process of ".xxx.typegen" file when we use `tsTypes` property? I'm currently trying to find the reason why the auto generated .typegen file is not generated inside the [nx's](https://nx.dev/) monorepo setup. I've tried the same exact machine configuration in another project and the .typegen file is generated automatically.

**Comment by Andarist** [comment]
This is somewhat weird as it shouldnt be dependent on a specific project configuration. Are you using VS Code extension to generate typegen files? Have you tried to use the `@xstate/cli` to do it?

**Reply by livingmine** [reply]
Yes I'm using the XState VSCode extension. The typegen is successfully generated when using the @xstate/cli. Any idea why it is not auto generated?
